### PR TITLE
feat: python bindings

### PR DESCRIPTION
**Pull request by nuberchardzer1**
- Added Python bindings for ZXC compressor/decompressor
- Exposed compress, decompress, stream_compress, stream_decompress
- Stream API works with file-like objects and releases the GIL

### CODE REVIEW

File: wrappers/python/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[build-system]
+requires = ["scikit-build-core>=0.9"]
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "zxc"
```
Comment (hellobertrand) [inline wrappers/python/pyproject.toml]: zxc is not available on pypi. Please using `python-zxc` if possible.

**Review by hellobertrand — CHANGES_REQUESTED** [review]
First off, thank you for taking the time to contribute python bindings to zxc. The implementation is nearly ready to merge! I've identified a few items that need attention. Please don't be discouraged by the detailed comments below. Most of the changes are minor, and the foundational work you've done is solid.

Please consider implementing `get_decompressed_size()` wrapper.

Please fixes inconsistent level default between files

Please add unit tests covering at least:
- Round-trip tests
- All compression levels
- Stream operations
- Edge cases: empty input, very large input (>1MB),  random incompressible data
- Error handling: invalid compressed data, corrupt checksums.

**Comment by hellobertrand** [comment]
To keep the project's legal and attribution standards consistent, could you please add the standard ZXC license header to the top of the new files? I’ve just updated the CONTRIBUTING.md with the official format to make it easier for everyone.

You can just copy-paste this block:
```
/*
 * ZXC - High-performance lossless compression
 *
 * Copyright (c) 2025-2026 Bertrand Lebonnois and contributors.
 * SPDX-License-Identifier: BSD-3-Clause
 */
```

**Comment by nuberchardzer1** [comment]
Hi! I've added your patches and merged all the commits to make it clean.

**Review by hellobertrand — APPROVED** [review]
Thanks a lot. I'll add a commit (on main) to update the README binding sections with your github username.

**Comment by hellobertrand** [comment]
I'll publish python-zxc in the next few days; I have a GitHub action to prepare.
